ingestion/src/metadata/ingestion/source/database/exasol/connection.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional
+from urllib.parse import quote_plus
+
+from pydantic import SecretStr
+from sqlalchemy.engine import Engine
+
+from metadata.generated.schema.entity.automations.workflow import (
+    Workflow as AutomationWorkflow,
+)
+from metadata.generated.schema.entity.services.connections.database.exasolConnection import (
+    ExasolConnection,
+)
+from metadata.ingestion.connections.builders import (
+    create_generic_db_connection,
+    get_connection_args_common,
+)
+from metadata.ingestion.connections.test_connections import test_query
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+def get_connection_url(connection: ExasolConnection) -> str:
```

Comment:
we can add a test similar to https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/tests/unit/test_source_connection.py#L1098

Reply:
Thanks for the link! I'll add a similar test.

Do you have any suggestions or requirements regarding integration tests?

Reply:
Integration test should not be needed, since most of the connector is based on the "core" implementation which is covered already. thanks